app/src/main/java/ch/redacted/ui/collage/CollagePresenter.java

```diff
@@ -0,0 +1,48 @@
+package ch.redacted.ui.collage;
+
+import javax.inject.Inject;
+
+import ch.redacted.data.DataManager;
+import ch.redacted.data.model.Collage;
+import ch.redacted.ui.base.BasePresenter;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.disposables.CompositeDisposable;
+import io.reactivex.observers.DisposableSingleObserver;
+import io.reactivex.schedulers.Schedulers;
+
+public class CollagePresenter extends BasePresenter<CollageMvpView> {
+
+    private final DataManager mDataManager;
+    private CompositeDisposable mSubscription = new CompositeDisposable();
+
+    @Inject
+    CollagePresenter(DataManager dataManager) {
+        mDataManager = dataManager;
+    }
+
+    void loadCollage(int id) {
+        checkViewAttached();
+        getMvpView().showLoadingProgress(true);
+
+        mSubscription.add(mDataManager.getCollage(id)
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribeOn(Schedulers.io())
+                .subscribeWith(new DisposableSingleObserver<Collage>() {
+                    @Override
+                    public void onSuccess(Collage collage) {
+                        //  @todo Bug: Can throw Null Pointer Exception?
+                        //  Caused by: java.lang.NullPointerException: Attempt to invoke interface method 'void ch.redacted.ui.collage.CollageMvpView.showCollage(ch.redacted.data.model.Collage)' on a null object reference
```

Comment:
This could be due to the android lifecycle. It is possible for the presenter to exist without the view (after rotation or recreation of the view, e.g. after returning to the app). Its mostly a flaw with the MVP architecture and holding a direct reference to the view. I wouldn't worry about it too much unless it happens very frequently.